src/nvim/ex_cmds.c

```diff
@@ -6028,6 +6089,8 @@ static buf_T *show_sub(exarg_T *eap, pos_T old_cusr, char_u *pat, char_u *sub,
   char_u *save_shm_p = vim_strsave(p_shm);
   size_t sub_size = mb_string2cells(sub);
   size_t pat_size = mb_string2cells(pat);
```

Comment:
This was another case where I apparently hyper-corrected from strlen() to mb_string2cells() in c04ffe866d276d6a6bd9e9c6a8b0dbb71504db7c . I think it may not be correct here, also. At least when `pat_size` is an argument to `bufhl_add_hl()` (below), that seems wrong. But it's also used to compute column widths in the "highlight the replaced part" section below ...

Reply:
Those are not used for anything anymore other than the decision if we need to show the split window, see https://github.com/neovim/neovim/pull/7231/files#diff-b9d70f1e57978ae9fbf83f73315b0c0fR6105.  We only check if those are 0 or not, so `mb_string2cells` or `strlen` should not matter (actually, we could really check `*sub` and `*pat`).

I'm not sure why that check is there anyways. If `pat` or `sub` are empty or `NULL`, doesn't that just mean the user left it out and the last recent pattern/substitution is used? Why wouldn't we want to show the preview window then?

I think we could just remove those 2 lines, remove this part of the check and remove those arguments from the function call.

Reply:
> I'm not sure why that check is there anyways. If pat or sub are empty or NULL, doesn't that just mean the user left it out and the last recent pattern/substitution is used?

Well, it's a matter of whether we think the user wants to see a preview of `:%s/`. At the time I guess this would be more annoying than helpful.

> I think we could just remove those 2 lines, remove this part of the check and remove those arguments from the function call.

👍  Let's put it in a separate commit for clarity.

Reply:
Hmm, right, maybe the user doesn't want that preview... I'll ponder it a bit and prepare a new commit tomorrow, this should be streamlined no matter what.

Reply:
I think the simplification is worth it.